hacspec/src/vec_integers_public.rs

```diff
@@ -0,0 +1,187 @@
+#![allow(unused_variables)]
+use crate::prelude::*;
+
+// TODO: drop the `Copy` requirement for matrices
```

Comment:
Ooooh Ok that's because you want to extend to matrices that you want to drop the `Copy`. And also vectors whose size is not known at compilation time I guess.

Reply:
Right. `Copy` limits us a little too much here. It's not great though.. As you see in the examples its doable without too much uglyness if we provide a version of the traits that implement `Copy`.